### PR TITLE
[Merged by Bors] - Add `Clone` trait to DefaultProduceRequest

### PR DESCRIPTION
**Pull request by galibey**
This is needed for being able to retry produce requests if an error occurs.

Related #2481

### CODE REVIEW

File: crates/fluvio-dataplane-protocol/src/batch.rs [hunk 1]
```diff
@@ -185,7 +184,7 @@ impl TryFrom<Batch> for Batch<RawRecords> {
         let compression = f.get_compression()?;
         let compressed_records = compression.compress(&buf)?;
```
Comment (morenol) [inline crates/fluvio-dataplane-protocol/src/batch.rs:185]: I think that we can make `compress()` to return `Bytes` instead of `Vec<u8`

Comment (galibey) [reply crates/fluvio-dataplane-protocol/src/batch.rs]: Great idea. Added it.

**Review by morenol — CHANGES_REQUESTED** [review]
LGTM, just need to bump apropriately

File: crates/fluvio-compression/Cargo.toml [hunk 2]
```diff
@@ -18,3 +18,4 @@ flate2 = { version = "1.0.20"}
 snap = { version = "1" }
 lz4_flex = { version = "0.9", default-features = false, features = ["safe-decode", "safe-encode", "frame"] }
 thiserror = "1.0.30"
+bytes = { version = "1.1.0"}
```
[the remark this answers is not in the document]
Comment (galibey) [reply crates/fluvio-compression/Cargo.toml:21]: Done

**Review by morenol — APPROVED** [review]
lgtm!

**Review by sehz — APPROVED** [review]
LGTM

**Comment by morenol** [comment]
bors retry

**Pull request closed without merging**